src/search/Search.jsx

```diff
@@ -159,6 +165,9 @@ class Search extends React.Component {
     };
 
     onSearch = () => {
+        if (this.props.disabled) {
```

Comment:
why not just remove binding if disabled? I mean like `<Button onSearch={disabled?null:this.onSearch}...`

Reply:
onSearch method is referred at more than one places. if i program like `<Button onSearch={disabled?null:this.onSearch}...`, it will  looks so guly.

Reply:
It would be better if you remove unnecessary binding, you can also keep the if statement in onSearch, just for double insurance

Reply:
> It would be better if you remove unnecessary binding, you can also keep the if statement in onSearch, just for double insurance

this solution will make the code not DRY(Donot Repeat Yourself)